### PR TITLE
#12551 Use nickname if available

### PR DESCRIPTION
**Pull request by jgandres**
Wherever a firstname / fullname is currently displayed, if the user has assigned to him/her a nickname, it should prevail. In example, default groupchat title, notifications about new messages, sender of a message, etc.

### CODE REVIEW

File: src/chatClient.cpp [hunk 1]
```diff
@@ -3789,6 +3850,88 @@ bool Client::isCallInProgress(karere::Id chatid) const
...
+{
+    std::string aliasBin;
+    std::map<uint64_t, std::string>::iterator it;
+    it = mAliasesMap.find(userId);
+    if (it != mAliasesMap.end())
+    {
+        std::string aliasB64 = it->second;
+        Buffer buf(aliasB64.size());
+        size_t decLen = base64urldecode(aliasB64.c_str(), aliasB64.size(), buf.buf(), buf.bufSize());
+        aliasBin.assign(buf.buf(), decLen);
```
Comment (sergiohs84) [inline src/chatClient.cpp]: All the above for a simple conversion from B64 to binary seems too complicated for a simple operation. Please, pay attention to my adjustment.

File: src/userAttrCache.cpp [hunk 2]
```diff
@@ -439,8 +428,14 @@ void UserAttrCache::fetchStandardAttr(UserAttrPair key, std::shared_ptr<UserAttr
...
-        item->data.reset(gUserAttrDescs[key.attrType].getData(*result));
-        item->resolve(key);
+
+        UserAttrDescMap::iterator it = gUserAttrDescsMap.find(key.attrType);
+        if (it != gUserAttrDescsMap.end())
+        {
+            auto& desc = it->second;
+            item->data.reset(desc.getData(*result));
+            item->resolve(key);
+        }
```
Comment (sergiohs84) [inline src/userAttrCache.cpp]: Is it possible to fetch an attribute that it's not at the `gUserAttrDescsMap`?? I think it shouldn't. Let's discuss about it.

**Comment by sergiohs84** [comment]
trigger compilation

File: src/userAttrCache.cpp (hunk 2 above)
Comment (jgandres) [reply src/userAttrCache.cpp]: I agree with you, we shouldn't fetch an attribute out of gUserAttrDescsMap

Comment (sergiohs84) [reply src/userAttrCache.cpp]: Then I'll simplify it :)

**Comment by sergiohs84** [comment]
TODO:
- Implement the deletion of aliases in `updateAliases()`.
- Add the _fullname_ to the contact in the contact list of the QtApp (if there's an alias).

**Pull request merged**